Firestore/core/test/unit/model/mutation_test.cc

```diff
@@ -712,6 +712,25 @@ TEST(MutationTest, OverlayWithFieldDeletionOfNestedField) {
   VerifyOverlayRoundTrips(doc, {patch1, patch2, patch3});
 }
 
+// See: https://github.com/firebase/firebase-ios-sdk/issues/9985
+TEST(MutationTest, OverlayWithFieldDeletionOfNestedFieldAndParentField) {
+  MutableDocument doc = Doc("collection/key", 1, Map("foo", 1));
+  Mutation patch1 =
+      PatchMutation("collection/key", Map("foo", "foo-patched-value"),
+                    {{"bar.baz", Increment(1)}});
+  Mutation patch2 =
+      PatchMutation("collection/key", Map("foo", "foo-patched-value"),
+                    {{"bar.baz", ServerTimestamp()}});
+  Mutation patch3 =
+      PatchMutation("collection/key", Map("foo", "foo-patched-value"),
+                    {Field("bar.baz")}, {});
+
+  Mutation patch4 = PatchMutation(
+      "collection/key", Map("foo", "foo-patched-value"), {Field("bar")}, {});
+
+  VerifyOverlayRoundTrips(doc, {patch1, patch2, patch3, patch4});
```

Comment:
can we also try multiple levels as well? e.g. `foo.bar.baz` ?

Reply:
Done.